### PR TITLE
Iteration four 

### LINKED ISSUE #15: Iteration 4: Sleep Class
- [x] Sleep class takes in user as a parameter (identified by their userID), the average number of hours slept per day
- [x] For a user, their average sleep quality per day over all time
- [x] For a user, how many hours they slept for a specific day (identified by a date)
- [x] For a user, their sleep quality for a specific day (identified by a date)
- [x] For a user, how many hours slept each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] For a user, their sleep quality each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] For all users, the average sleep quality

### LINKED ISSUE #16: Iteration 4: Sleep Class Tests
### Will verify that:

- [x] Sleep should be a function
- [x] Sleep should take in a user id as a parameter
- [x] Sleep should calculate the average number of hours slept per day
- [x] Sleep should calculate their average sleep quality per day over all time
- [x] Sleep should calculate how many hours they slept for a specific day (identified by a date)
- [x] Sleep should calculate their sleep quality for a specific day (identified by a date)
- [x] Sleep should calculate how many hours slept each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] Sleep should calculate their sleep quality each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] Sleep should calculate the average sleep quality

### PR DESCRIPTION
**Pull request by Andrew-Musselman**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
    - Changed: Added methods for sleep data - daily average quality and quantity, weekly view/array for quality and quantity
    - Bugs created: 
        - Possible bug/problem with calculating the total average amount of sleep for all users
        - Possible merge conflict as well
- [ ] Bug fix (non-breaking change which fixes an issue)
    -Bug fixed:
- [x] Refactor
    -Code refactored: DRYed up some of the methods
- [x] Documentation update: 
    - Edited some of the tests to utilize the correct datasets and deep equal instead of equal for complex data

## How Has This Been Tested?

Mocha & Chai unit testing. 

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have fully tested my code
- [x] My code follows the style guidelines of this project.
- [x] My code has no unused/commented-out code.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

## Future Steps:

Add future steps or next steps here.

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Sleep{
+  constructor(data, id) {
```
Comment (Andrew-Musselman) [inline src/Sleep.js:2]: Since we were filtering by idea for our sleep class, I took out the filter by id in the User methods to DRY it up a bit

**Comment by Andrew-Musselman** [comment]
Haha, I just noticed the merge conflict is in the README...

File: src/Sleep.js (hunk 1 above)
Comment (zliibbe) [reply src/Sleep.js:2]: That sounds like a great idea. I don't think we need the ID here.

File: src/User.js [hunk 2]
```diff
@@ -36,6 +36,118 @@ class User {
...
+
+  getSevenDaysOfSleepQuality(sleepData, dates) {
+    return sleepData.map(userObj => {
+      dates.includes(userObj.date)
+      return userObj.sleepQuality
+    })
+  }
+
+
+  //input: sleepData & date range- array of date strings ><
```
Comment (BrendaSotel0) [inline src/User.js]: Lines 88-142 can be deleted now.

**Review by zliibbe — APPROVED** [review]
These changes make sense and as long as code is functional and tested we can merge following clean up of comments.

**Review by BrendaSotel0 — COMMENTED** [review]
Looking good

**Pull request merged**